storage/s3/options.go

```diff
@@ -87,30 +221,13 @@ func NewOptions(opts ...*Options) *Options {
 		BufferSize: &defaultBufferSize,
 	}
 	for _, opt := range opts {
-		if opt.StaticCredentials != nil {
-			ret.StaticCredentials = opt.StaticCredentials
-		}
-		if opt.Region != nil {
-			ret.Region = opt.Region
+		ret.storageSettings.patch(&opt.storageSettings)
+		if opt.DefaultExpire != nil {
+			ret.DefaultExpire = opt.DefaultExpire
```

Comment:
this is not in the `patch` on purpose, right?

Reply:
Patch applies to the `storageSettings` subset of the `Options`.